ROADMAP.md

```diff
@@ -1,31 +1,42 @@
 # Roadmap
 
-This document defines a high level roadmap for Rook development. The dates below are subject to change but should give a general idea of what we are planning. We use the [milestone](https://github.com/rook/rook/milestones) feature in Github so look there for the most up-to-date and issue plan.
+This document defines a high level roadmap for Rook development.
+The dates below are subject to change but should give a general idea of what we are planning.
+We use the [milestone](https://github.com/rook/rook/milestones) feature in Github so look there for the most up-to-date and issue plan.
 
-## Rook 0.7
+## Rook 0.8
 
-- Ceph Block Storage (Cluster and Pool CRDs) declared beta
-- Mon reliability (restarts, failing over too fast, ip changes, etc.)
-- Durability of state (local storage support, config is regenerat-able)
-- Run everywhere (rook-agent uses ceph-fuse, nbd-rbd / tcmu runner)
-- Adding / removing nodes and disk drives (lifecycle issues, failures, etc.)
-- Run with Least Privileged and possibly without privileged containers
-- Shutdown / restart issues
-- Improved data placement and pool configuration (CRUSH maps)
-- Placement group balancer support (ceph-mgr module)
-- Use new Prometheus ceph-mgr module
-- Performance Testing for Block storage
-- Make API and CLI Optional (plan to remove)
-- Dynamic Volume Provisioning for CephFS
+* Multiple storage backends
+  * Full design
+  * Refactor code base and repositories to enable
+  * Consider support for Minio, potentially early support for other backends time permitting
+* Switch CRDs to API Aggregation
+* Run on arbitrary PVs
+* Remove Rook API and CLI
+* Migrate CI and release pipelines to a solution hosted by the CNCF
+* Run with Least Privileged and possibly without privileged containers
+* Shutdown / restart issues
+* Support Kubernetes 1.7+ only
+* Ceph features and improvements
+  * Adding/removing disks (lifecycle issues, failures, 1 OSD per pod, etc.)
+  * Placement group balancer support (ceph-mgr module)
+  * Mon reliability (restarts, failing over too fast, ip changes, etc.)
+  * Mimic support
 
-## Rook 0.8
+## Rook 0.9
+
+* Automated upgrade
+* Operator runs HA
+* Durability of state (local storage support, config can be regenerated)
+* Custom resource validation, progress, status
+* Design for Volume Snapshotting and policies (consider aligning with SIG-storage)
+* Ceph features and improvements
+  * Improved data placement and pool configuration (CRUSH maps)
+  * Dynamic Volume Provisioning for CephFS
+  * ceph-volume provisioning for OSDs
+  * Ceph CSI plug-in that runs everywhere (rook-agent uses ceph-fuse, nbd-rbd / tcmu runner)
+* Cluster, Pool, and VolumeAttach types are declared Beta
```

Comment:
I think VolumeAttach should be deprecated and not declared Beta. K8S already has a `VolumeAttachment` object that was built as part of the CSI support.  See here: https://github.com/kubernetes/api/blob/master/storage/v1alpha1/types.go and this object is moving to beta on 1.10: https://github.com/kubernetes/kubernetes/pull/58462.

Rook's VolumeAttach was created to support flexvolume in the absent of a resource of that type in kubernetes. But with Rook moving to CSI, i dont think there would be an use case for it.

Reply:
do you think it's possible or would it make sense to use the official VolumeAttach type before  we convert our flexvolume to CSI?  I'm guessing that would not be a worthwhile effort.

I assume we are going to use the official VolumeAttach type later on with CSI (and continue to use our own VolumeAttach with flexvolume in the meantime).  In that case, you are probably correct that declaring our VolumeAttach to be Beta is not useful since we don't intend for it to be a long lived type.